src/Hal/Report/Cli/Reporter.php

```diff
@@ -4,19 +4,23 @@
 use Hal\Application\Config\Config;
 use Hal\Component\Output\Output;
 use Hal\Metric\Consolidated;
+use Hal\Metric\Metric;
 use Hal\Metric\Metrics;
-
-class Reporter
+use Hal\Report\ReporterInterface;
+use function array_slice;
+use function arsort;
+use function round;
+use function sprintf;
+
+/**
+ * This class takes care about the global output into the STDOUT of consolidated metrics.
+ */
+final class Reporter implements ReporterInterface
```

Comment:
making it final wont it be a BC?

Reply:
I'm not in favor of final for OSS code -> we never know how code will be used

Reply:
Removed `final` because of potential BC break.